components/gfx/text/text_run.rs

```diff
@@ -304,6 +304,21 @@ impl<'a> TextRun {
         })
     }
 
+    /// Returns the index in the range of the first glyph advancing over given advance
+    pub fn range_index_of_advance(&self, range: &Range<ByteIndex>, advance: Au) -> usize {
+        // TODO(Issue #199): alter advance direction for RTL
+        // TODO(Issue #98): using inter-char and inter-word spacing settings when measuring text
+        let mut remaining = advance;
+        self.natural_word_slices_in_range(range)
+            .map(|slice| {
+                let (slice_index, slice_advance) =
+                    slice.glyphs.range_index_of_advance(&slice.range, remaining, self.extra_word_spacing);
+                remaining -= slice_advance;
```

Comment:
@pcwalton `glyphs.range_index_of_advance` now returns `(index, advanced)` instead of `(index, remaining)` to make it more intuitive. Therefore we need to update `remaining` here.